Sources/PackageModel/Target.swift

```diff
@@ -11,12 +11,53 @@
 import Basic
 
 public class Target: ObjectIdentifierProtocol {
+
+    /// The type of library.
+    public enum LibraryType: String, Codable {
```

Comment:
Hmm, I am not sure this is correct way to model this. Targets are not static/dynamic/automatic, products are.

Reply:
How do you recommend modelling this? The generate-xcodeproj uses the `Target` to generate its dependencies ...

https://github.com/apple/swift-package-manager/blob/master/Sources/Xcodeproj/Target+PBXProj.swift#L38-L48